use descriptive link text for the Advanced SnarkyJS topics (#401)
o1-labs/docs2: docs/zkapps/advanced-snarkyjs/actions-and-reducer.mdx

The `acc` shown earlier is now `state`; you also have to pass in the state's type as a parameter. In addition, you have to pass in an `actionState` which refers to one particular point in the action's history. Like `Array.reduce`, `Reducer.reduce` takes a callback that has the signature `(state: S, action: A) => S`, where `S` is the `stateType` and `A` is the `actionType`. It returns the result of applying all the actions, in order, to the initial `state`. In this example, the returned `state` is `Bool(true)` because one of the actions in the list is `Field(1000)`. Reduce also returns the new actionState -- so you can store it to use when you reduce the next batch of actions. One last difference to JS reduce is that this takes a _list of lists_ of actions instead of a flat list. Each of the sublists are the actions that were dispatched in one account update (for example, while running one smart contract method).

> As an astute reader, you might have noticed that this use of `state` is eerily similar to a standard "Elm architecture" because an instance of a scan over an implicit infinite stream of actions (though here that are aggregated in chunks). This problem is familiar to web developers through its instantiation by using the Redux library or more recently by using the `useReducer` hook in React.

Author: removed the link to a 2018 blog post

> As an astute reader, you might have noticed that this use of `state` is eerily similar to a standard "Elm architecture" because an instance of a scan over an implicit infinite stream of actions (though here that are aggregated in chunks). This problem is familiar to web developers through its instantiation by using the Redux library or more recently by using the `useReducer` hook in React.

Reviewer: ```suggestion
As an astute reader, you might have noticed that this use of `state` is eerily similar to a standard "Elm architecture" that scans over an implicit infinite stream of actions (though here they are aggregated in chunks). This problem is familiar to web developers through its instantiation by using the Redux library or more recently by using the `useReducer` hook in React.
```
